crypto/err/err.c

```diff
@@ -127,8 +127,8 @@ static LHASH_OF(ERR_STRING_DATA) *int_error_hash = NULL;
 static int int_err_library_number = ERR_LIB_USER;
 
 static unsigned long get_error_values(int inc, int top, const char **file,
```

Comment:
It would be great if ```inc``` and ```top``` could be merged into a single enum.  I never got around to figuring out how to do that cleanly.  hint, hint.

Reply:
There is no way.  The error queue is a circular buffer, and to handle those, you *must* know where it starts and when it stops.

Reply:
No, take a look at how those two parameters are are used:
```
; grep get_error_values crypto/err/err.c
static unsigned long get_error_values(int inc, int top, const char **file,
    return get_error_values(1, 0, NULL, NULL, NULL, NULL);
    return get_error_values(1, 0, file, line, NULL, NULL);
    return get_error_values(1, 0, file, line, data, flags);
    return get_error_values(0, 0, NULL, NULL, NULL, NULL);
    return get_error_values(0, 0, file, line, NULL, NULL);
    return get_error_values(0, 0, file, line, data, flags);
    return get_error_values(0, 1, NULL, NULL, NULL, NULL);
    return get_error_values(0, 1, file, line, NULL, NULL);
    return get_error_values(0, 1, file, line, data, flags);
static unsigned long get_error_values(int inc, int top, const char **file,
```
They are *flags* and at most one of them should be set; look at the if statement around line 432.

Reply:
Oh that. Sorry, misunderstood...
As for a single enum, I dunno. What names would they get?
Considering this is highly internal, I'm currently leaning toward "Oh bother...", but if someone comes up with a sensible scheme, then sure why not...

Reply:
I think it could really help make the code more clear, and avoid a run-time error condition, but it's not the most important thing in the world.

Reply:
Yeah, slightly useful.  #9870